src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -904,21 +904,36 @@ protected static void InlineParentClasses(CodeElement currentElement, CodeElemen
                     .Properties
                     .Where(pp =>
                         !currentClass.ContainsMember(pp.Name) &&
-                        !currentClass.Properties.Any(cp => cp.Name.Equals(pp.Name, StringComparison.OrdinalIgnoreCase)));
+                        !currentClass.Properties.Any(cp => cp.Name.Equals(pp.Name, StringComparison.OrdinalIgnoreCase)))
+                    .Select(p =>
```

Comment:
those are "improvements" over the first iteration.

Reply:
while this technically works, select is meant for projections. I'm not sure what the implications of hijacking linq this way are.  What would you think about building a union of all those subsets and iterating (foreach) over that do make the change instead?

Reply:
Proposed a different encoding, let me know if this works for you! 👍 

(feeling curious about "hijacking linq", I was assuming `Select` to have the same semantics of [Map](https://reactivex.io/documentation/operators/map.html), but happy to learn more!)

Reply:
as far as I understand projection methods are not meant to be used to mutate the data. This impacts the ability to vectorialize them and other aspects (the result of a select might be called multiple times as it's IEnumberable, in which case the alloc might be performed multiple times)